### PR TITLE
Video: link to fallback player for dance-2019 videos

### PR DESCRIPTION
**Pull request by breville**
If we're on a `dance-2019` level, and we don't already have a `force_youtube_fallback` parameter in the URL, then show an extra link in the popup video header which will add `force_youtube_fallback=1` to the URL parameters and reload the page.

Screenshot 2019-12-06 14 19 08

### CODE REVIEW

**Comment by uponthesun** [comment]
Josh Schulte had some feedback on wording in the slack thread, can you work that out with him?

File: apps/src/code-studio/videos.js [hunk 1]
```diff
@@ -202,6 +202,25 @@ videos.showVideoDialog = function(options, forceShowVideo) {
   var nav = $div.find('.ui-tabs-nav');
   nav.append(download);
 
+  // Possibly append a switch to fallback player.
+  if (
+    window.location.href.includes('/s/dance-2019') &&
+    !window.location.href.includes('force_youtube_fallback')
```
Comment (islemaster) [inline apps/src/code-studio/videos.js:208]: Nonblocking: Could search particular parts of the URL.

```js
    window.location.pathname.includes('/s/dance-2019') &&
    !window.location.search.includes('force_youtube_fallback')
```

File: apps/src/code-studio/videos.js [hunk 2]
```diff
@@ -274,6 +293,33 @@ videos.showVideoDialog = function(options, forceShowVideo) {
...
+  var newParam = key + '=' + val,
+    params = '?' + newParam;
+
+  // If the "search" string exists, then build params from it
+  if (search) {
+    // Try to replace an existance instance
+    params = search.replace(
+      new RegExp('([?&])' + key + '[^&]*'),
+      '$1' + newParam
+    );
```
Comment (islemaster) [inline apps/src/code-studio/videos.js:312]: Nonblocking for this use case: This function fails in the case where one queryparam key is a subset of another; trying to replace `part1=foo` might replace `part10=bar` instead if it showed up first in the search string.

**Review by islemaster — APPROVED** [review]
Tested locally.  This looks great!

**Pull request closed without merging**